### PR TITLE
[ODS-4505] Fix config transformation to address postgres build

### PR DESCRIPTION
**Pull request by tbmeador-dlp**

### CODE REVIEW

**Review by jamessmckay — APPROVED** [review]

File: Application/EdFi.Common/Configuration/DatabaseEngine.cs [hunk 1]
```diff
@@ -21,6 +21,17 @@ public DatabaseEngine(string value, string displayName, string scriptsFolderName
         }
 
         public string ScriptsFolderName { get; private set; }
+
+        public static DatabaseEngine TryParseEngine(string value)
```
Comment (blmeyers) [inline Application/EdFi.Common/Configuration/DatabaseEngine.cs:25]: If we want an exception why not just use Parse?  Does it have to be this message and a NotSupportedException?

File: Application/EdFi.Ods.Common/Configuration/ApiSettings.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public class ApiSettings
 
         public ApiSettings()
         {
-            _databaseEngine = new Lazy<DatabaseEngine>(() => DatabaseEngine.FromValue(Engine));
+            _databaseEngine = new Lazy<DatabaseEngine>(() => DatabaseEngine.TryParseEngine(Engine));
```
Comment (blmeyers) [inline Application/EdFi.Ods.Common/Configuration/ApiSettings.cs:21]: Should this be outside of a Lazy so it fails immediately for bad configuration?

File: Application/EdFi.Security.DataAccess/Contexts/PostgresSecurityContext.cs [hunk 3]
```diff
@@ -9,9 +9,22 @@
 using System.Data.Entity.ModelConfiguration.Conventions;
 using EdFi.Ods.Common;
 using EdFi.Ods.Common.Utils.Extensions;
+using Npgsql;
 
 namespace EdFi.Security.DataAccess.Contexts
 {
+    class NpgsqlConfiguration : DbConfiguration
```
Comment (blmeyers) [inline Application/EdFi.Security.DataAccess/Contexts/PostgresSecurityContext.cs:16]: See previous comment about making this in EdFi.Common.

**Review by blmeyers — CHANGES_REQUESTED** [review]

File: Application/EdFi.Common/Configuration/DatabaseEngine.cs (hunk 1 above)
Comment (tbmeador-dlp) [reply Application/EdFi.Common/Configuration/DatabaseEngine.cs:25]: Here I was just following the same pattern as the `ApiMode`

**Review by blmeyers — APPROVED** [review]

**Pull request merged**